MapboxCoreNavigation/RouteController.swift

```diff
@@ -854,19 +857,20 @@ extension RouteController: CLLocationManagerDelegate {
         }
 
         routeProgress.currentLegProgress.currentStepProgress.userDistanceToManeuverLocation = userAbsoluteDistance
-
-        guard let spokenInstructions = routeProgress.currentLegProgress.currentStepProgress.remainingSpokenInstructions else {
-            print("The directions request was made without `includesVoiceInstructions` enabled. This will prevent users from getting voice instructions. It's recommended to make your directions request via `NavigationRouteOptions()`.")
-            return
-        }
-
+    }
+    
+    func updateSpokenInstructionProgress(for location: CLLocation) {
+        guard let coordinates = routeProgress.currentLegProgress.currentStep.coordinates else { return }
+        let userSnapToStepDistanceFromManeuver = Polyline(coordinates).distance(from: location.coordinate)
```

Comment:
We’re calculating this distance at least three other times on each location update:

https://github.com/mapbox/mapbox-navigation-ios/blob/ebcfcea8bb2502c0f102dca226c8fc022fff96cd/MapboxCoreNavigation/RouteController.swift#L564
https://github.com/mapbox/mapbox-navigation-ios/blob/ebcfcea8bb2502c0f102dca226c8fc022fff96cd/MapboxCoreNavigation/RouteController.swift#L826
https://github.com/mapbox/mapbox-navigation-ios/blob/ebcfcea8bb2502c0f102dca226c8fc022fff96cd/MapboxCoreNavigation/RouteController.swift#L588

Can we consolidate this code into a single calculation that gets persisted on the route progress somehow?